themes/powerline-icon/powerline-icon.base.sh

```diff
@@ -41,3 +41,32 @@ function __powerline_last_status_prompt {
     echo "✅|${LAST_STATUS_THEME_PROMPT_COLOR_SUCCESS}"
   fi
 }
+
+function __powerline_prompt_command {
+  local last_status="$?" ## always the first
+  local separator_char="${POWERLINE_PROMPT_CHAR}"
+
+  LEFT_PROMPT=""
+  SEGMENTS_AT_LEFT=0
+  LAST_SEGMENT_COLOR=""
+
+  # The IFS (internal field seperator) may have been changed outside to not contain
+  # the space character ' ' whence we need to make sure that the space separated list
+  # stored in POWERLINE_PROMPT is converted into an array correctly.
+  IFS=' ' read -r -a POWERLINE_PROMPT_ARRAY <<< "${POWERLINE_PROMPT}"
+
+  ## left prompt ##
+  for segment in ${POWERLINE_PROMPT_ARRAY[@]}; do
+    local info="$(__powerline_${segment}_prompt)"
+    [[ -n "${info}" ]] && __powerline_left_segment "${info}"
+  done
+  __powerline_left_segment $(__powerline_last_status_prompt ${last_status})
```

Comment:
Let's start the discussion. I rather think this should be fixed in `powerline.base.sh`. For the other segments, we first assign the results to the variable `info`, and only when the variable is no-empty, we call `__powerline_left_segment`. We can follow the same procedure in `powerline.base.sh`.

Reply:
Right.

Reply:
Only to see if I understood correctly, the function called on line 63 should only be invoked if the 'info' variable for the segments is empty. However, in the original line, this function is executed only if the command executed by the user does not return a status of 0. Should I correct something specific in this line?

Reply:
Please modify that line in `powerline.base.sh`. As I have written, first assign the result of `$(__powerline_last_status_prompt ${last_status})` to the variable `info`. Then instead of checking the `last_status`, please check that `info` has a non-empty value. If yes, call `__powerline_left_segment` with the argument being `"$info"`.